### DISCUSSION TITLE
Cannot overwrite the Icon of a ComboBox

### DISCUSSION THREAD

**Discussion by Patinator01** [AvaloniaUI/Avalonia#12069] [Q&A]
Hi all,

i am currently trying to style my ComboBox. Therefore i also want to change the dropdown icon but i am still not able to change it via a Style. Is it even possible to override it like this? Oder do i have to write my own template.

`<UserControl xmlns="https://github.com/avaloniaui"
             xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
             xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
             xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
             xmlns:styles="clr-namespace:Material.Styles;assembly=Material.Styles"
             mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="450"
             x:Class="PelviPower.Screens.Settings.SettingsView">

    <UserControl.Styles>
        <Style Selector="ComboBox ToggleButton /template/ Path#arrow">
            <Setter Property="Fill" Value="red" />
           <Setter Property="Data" Value="M4 0 L8 8 L0 8 L4 0" />
        </Style>
    </UserControl.Styles>

    <Grid RowDefinitions="Auto,Auto" HorizontalAlignment="Center" VerticalAlignment="Center">
        <Grid.ColumnDefinitions>
            <ColumnDefinition Width="*" MinWidth="300" />
            <ColumnDefinition Width="100" />
            <ColumnDefinition Width="*" MinWidth="300" />
        </Grid.ColumnDefinitions>

        <TextBlock Grid.Row="0" Grid.Column="0" Classes="ComboBoxHeader" Text="{DynamicResource Settings.Language}" />
        <ComboBox Grid.Row="1" Grid.Column="0" Items="{Binding PossibleLanguages}"
                  SelectedItem="{Binding SelectedLanguage, Mode=TwoWay}" />

        <TextBlock Grid.Row="0" Grid.Column="2" Classes="ComboBoxHeader" Text="{DynamicResource Settings.SystemType}" />
        <ComboBox Grid.Row="1" Grid.Column="2" Items="{Binding PossibleSystemTypes}"
                  SelectedItem="{Binding SelectedSystemType, Mode=TwoWay}" />
    </Grid>
</UserControl>
`

Currently it looks like this:
![image](https://github.com/AvaloniaUI/Avalonia/assets/44290347/e4544c9b-8eba-483c-8bb4-2ec9a24035b9)

But i want to let it look like this:
![image](https://github.com/AvaloniaUI/Avalonia/assets/44290347/5edf67f6-fa7a-4177-9cd4-fa5d3e2cd428)

Has someone an idea why it does not work?

Thanks for your help :)

**Comment by timunie** [comment]
which avalonia version? 

In general, please read this guide: https://docs.avaloniaui.net/docs/next/basics/user-interface/styling/troubleshooting

And also lookup original styles. Assuming you're using 11.0 it is here: 
https://github.com/AvaloniaUI/Avalonia/blob/9806f27f02b75f22c20514b8229c62f6550d0b0c/src/Avalonia.Themes.Fluent/Controls/ComboBox.xaml#L101-L112

As the path data is hard-coded here, you need to copy the template and edit it on your own. Maybe a good idea to provide path data via dynamic resource here, so styling it gets easier. 
@maxkatz6 any thoughts?

**Reply by robloo** [reply]
For me, it makes sense to create and share a `DropDownGlyphGeometry` lightweight styling resource. We already had to deviate from upstream here.

Otherwise, they absolutely can drill into the control template with a styling selector and change the geometry. 

Edit: Also keep in mind margins and padding may assume certain geometry measurements.
